src/cosmicqc/detection.py

```diff
@@ -0,0 +1,619 @@
+        return iqr > self.variability_threshold
+
+    def check_skew_and_variable(self) -> None:
+        """
+        Step 1: Check for skewness in the cytoplasm texture around the nucleus and
+        variability in the FormFactor of the nucleus.
+        """
+        print("Running step 1...")
+
+        # perform skewness and variability tests
+        self.is_skewed = self.skewness_test_cytoplasm_texture()
+        self.is_variable = self.variability_test_formfactor()
+
+        # format results in a pretty table
+        print("Summary:\n")
+        print(
+            tabulate(
+                [
+                    ["Texture skewed?", bool_to_emoji(self.is_skewed)],
+                    ["Nucleus shape variable?", bool_to_emoji(self.is_variable)],
+                ],
+                headers=["Check", "Result"],
+                tablefmt="simple",
+            )
+        )
+
+        # set interpretation based on results
+        if self.is_skewed and self.is_variable:
+            interpretation = (
+                "Contamination detected! 🚨\n"
+                "Anomalous texture around nuclei detected and "
+                "nuclei segmentation is impacted.\n"
+                "Proceeding to step 2..."
+            )
+        elif self.is_skewed:
+            interpretation = (
+                "Contamination detected! 🚨\n"
+                "Anomalous texture around nuclei detected but "
+                "nuclei segmentation not clearly impacted.\n"
+                "Proceeding to step 2..."
+            )
+        elif self.is_variable:
+            interpretation = (
+                "Potential contamination detected! 🛑\n"
+                "This could indicate contamination, segmentation issues, "
+                "or an interesting phenotype.\n"
+                "Proceeding to step 2..."
+            )
+        else:
+            interpretation = "No indication of contamination. Plate appears clean 🫧!"
+
+        print("Interpretation:\n")
+        print(interpretation)
+
+    def calculate_texture_mean(self) -> bool:
+        """
+        Check the mean value of cytoplasm texture around a nucleus feature to
+        determine if whole plate or partial plate contamination is present.
+        Threshold is set based on findings from multiple experiments
+        (NF1 & CFReT, see citation file).
+
+        Normal raw values of this feature range below 0. In a clean plate,
+        it is noted that the normal values range from -0.5 to -0.3.
+        We set the threshold to -0.25 to detect whole plate contamination.
+
+        Returns:
+            boolean: True if whole plate contamination is detected, False otherwise
+            indicates partial contamination.
+        """
+        # check if the mean value is above the threshold
+        # True = whole plate, False = partial plate
+        return self.dataframe[self.cyto_feature].mean() >= self.texture_mean_threshold
+
+    def calculate_formfactor_mean(self) -> bool:
```

Comment:
When building classes we can use the `_` underscore prefix for method names to indicate "privacy" - that a method is only intended to be used by the class and isn't designed for use outside of the class. Consider making use of this where appropriate to help indicate what might be internal functionality. For example, if you decide that `calculate_formfactor_mean` is internal-only, it would be `_calculate_formfactor_mean`.

Reply:
Added!